### DISCUSSION TITLE
Runtime Error while trying multi gpu training

### DISCUSSION THREAD

**Discussion by Ca-ressemble-a-du-fake** [coqui-ai/TTS#1867] [General Q&A]
Hi,

I am willing to use my 2 owned GTX 1660 super GPUs to do transfer learning from an English model. So following the [tutorial](https://tts.readthedocs.io/en/latest/faq.html#how-can-i-train-my-own-tts-model) (see step 6 although I could not find `distribute.py`) I started the training with the following command :

`
CUDA_VISIBLE_DEVICES="0,1" python3 -m trainer.distribute --script /home/caraduf/YourTTS/TTS/TTS/bin/train_tts.py --config_path /home/caraduf/dataset1/config_VITS_Dual.json --restore_path  model_file.pth.tar
`

But I get 
`
[1] is setting up NCCL communicator and retreiving ncclUniqueId from [0] via c10d key-value store by key '0', but store->get('0') got error: Broken pipe
`

Note that running `train_tts.py` on a single device works flawlessly witht the following command :
`
CUDA_VISIBLE_DEVICES="0" python3 /home/caraduf/YourTTS/TTS/TTS/bin/train_tts.py --config_path /home/caraduf/dataset1/config_VITS_Dual.json --restore_path  model_file.pth.tar
`

Where does the multi gpu issue I get come from ? Where should I look at ?

Thanks in advance for your help,

**Comment by Ca-ressemble-a-du-fake** [comment]
Actually `distribute.py` is in `utils` folder.

So trying the official command : `CUDA_VISIBLE_DEVICES="0,1" python3 /home/caraduf/YourTTS/TTS/TTS/utils/distribute.py --script /home/caraduf/YourTTS/TTS/TTS/bin/train_tts.py --config_path /home/caraduf/dataset1/config_VITS_Dual.json --restore_path  model_file.pth.tar`

Did not show anything (no error, nothing). The process exits and the prompt is shown again.

**Comment by erogol** [comment]
you should use the one from the Trainer. The issue is about your local NCLL setup but I'm not sure what would fix it.

**Reply by Ca-ressemble-a-du-fake** [reply]
Ok thank you @erogol. I will remove and reinstall everything from scratch (pytorch,  cuda,  drivers, pip packages,...).

**Reply by Ca-ressemble-a-du-fake** [reply]
I removed all pip packages, all nvidia packages, rebooted, reinstalled drivers, ran make system-deps, make install. But the same error remains. Using cuda 11.7 with drivers 515.65.01.

**Reply by Ca-ressemble-a-du-fake** [reply]
Actually the CUDA version used by torch (`torch.cuda.version`) was the 10.2 even though I had the latest Nvidia drivers. So I removed the installed version of torch as [advised by ptrblck](https://discuss.pytorch.org/t/nvidia-geforce-rtx-3090-with-cuda-capability-sm-86-is-not-compatible-with-the-current-pytorch-installation/141940/17) and installed pytorch with CUDA 11.3 following the [command on their website](https://pytorch.org/get-started/previous-versions/). Now torch reports that it uses CUDA 11.3 and training on RTX GPU is working. Then I'll put the 2 1660 back and see if it works!

**Reply by Ca-ressemble-a-du-fake** [reply]
Despite my best efforts the above error remains. NCCL has been installed following NVIDIA's instructions (20.04 for x86_64). Anyway with 4GB of RAM it would not have gone through the training process. So for now training my French model is postponed!
